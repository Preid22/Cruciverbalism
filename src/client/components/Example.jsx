import React from "react";
import PropTypes from "prop-types";

function Welcome(props) {
  const { name, message } = props;
  return (
    <div className="p-6 max-w-sm mx-auto bg-white rounded-xl shadow-lg flex items-center m-4">
      <div>
        <div className="text-xl font-medium text-black">Hello {name}</div>
        <p className="text-slate-500">{message}</p>
      </div>
    </div>
  );
}

Welcome.propTypes = {
  name: PropTypes.string,
  message: PropTypes.string,
};

Welcome.defaultProps = {
  name: "Stranger",
  message: "This is an example component!",
};

export default Welcome;
