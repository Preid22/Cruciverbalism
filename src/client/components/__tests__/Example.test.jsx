import renderer from "react-test-renderer";
import React from "react";
import Example from "../Example";

it("renders correctly", () => {
  const tree = renderer
    .create(<Example name="Shane" message="This is an example component" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
