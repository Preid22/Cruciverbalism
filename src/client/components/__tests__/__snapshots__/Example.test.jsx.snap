// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="p-6 max-w-sm mx-auto bg-white rounded-xl shadow-lg flex items-center m-4"
>
  <div>
    <div
      className="text-xl font-medium text-black"
    >
      Hello 
      Shane
    </div>
    <p
      className="text-slate-500"
    >
      This is an example component
    </p>
  </div>
</div>
`;
